packages/framework/collabspaces/src/collabSpaces.ts

```diff
@@ -455,42 +457,41 @@ export class TempCollabSpaceRuntime
 		return this.createCollabChannel(value, channelId);
 	}
 
+	private populateRowColMaps() {
+		const rowCount = this.matrix.rowCount;
+		const colCount = this.matrix.colCount;
+		for (let rowIdIndex = 1; rowIdIndex < rowCount; rowIdIndex++) {
+			const currentRowId = this.matrix.getCell(rowIdIndex, 0) as unknown as string;
+			this.rowMap.set(currentRowId, rowIdIndex);
+		}
+
+		for (let colIdIndex = 1; colIdIndex < colCount; colIdIndex++) {
+			const currentColId = this.matrix.getCell(0, colIdIndex) as unknown as string;
+			this.colMap.set(currentColId, colIdIndex);
+		}
+	}
+
 	private mapChannelToCell(channelId: string) {
 		const parts = channelId.split(",");
 		assert(parts.length === 3, "wrong channel ID");
 		const rowId = parts[0];
 		const colId = parts[1];
 		const iteration = parts[2];
 
-		// TBD(Pri0)
-		// This needs to be way more efficient by building a reverse mapping and maintaining it through all operations!
-		// Need to reconcile which layer does such mapping. At the moment, this is done outside of matrix - in Tablero
-		// component. We should either move it here (and ideally - in compatible way, i.e. not changing document schema),
-		// or move most of the channel logic to the layer that implements row/col stable IDs.
-
-		const rowCount = this.matrix.rowCount;
-		const colCount = this.matrix.colCount;
-
-		let row;
-		for (row = 1; row < rowCount; row++) {
-			if ((this.matrix.getCell(row, 0) as unknown as string) === rowId) {
-				break;
-			}
+		// In case the channel is about to be created we need to populate the rowMap and colMap maps.
+		if (this.rowMap.size === 0 || this.colMap.size === 0) {
+			this.populateRowColMaps();
 		}
-		assert(row !== rowCount, "channel not found");
 
-		let col;
-		for (col = 1; col < colCount; col++) {
-			if ((this.matrix.getCell(0, col) as unknown as string) === colId) {
-				break;
-			}
-		}
-		assert(col !== colCount, "channel not found");
+		const row = this.rowMap.get(rowId);
+		const col = this.colMap.get(colId);
 
+		assert(row !== undefined, "channel's row not found");
+		assert(col !== undefined, "channel's col not found");
```

Comment:
Please add validation that indeed cells[row, 0] === rowId, same for column.
It's a bit expensive (2 more reads from matrix, no easy way to produce debug only code in JS), but I'd rather pay that price now (and remove it later if profiles point to it) then to be sorry we did not find a bug sooner.